<?xml version="1.0"?>
<launch>
    <!-- Start your HoloOcean publisher node -->
    <node pkg="holoocean_ros" type="depth_sim_ros.py" name="holocean_publisher" output="screen"/>

    <!-- Start nodelet manager -->
    <node pkg="nodelet" type="nodelet" name="nodelet_manager" args="manager" output="screen"/>

    <!-- Start depth_image_proc nodelet -->
    <node pkg="nodelet" type="nodelet" name="depth_to_pointcloud" 
          args="load depth_image_proc/point_cloud_xyz nodelet_manager" output="screen">
        <remap from="image_rect" to="/camera/depth/image_raw"/>
        <remap from="camera_info" to="/camera/depth/camera_info"/>
        <remap from="points" to="/camera/depth/points"/>
        <param name="sensor_frame" value="camera_depth_optical_frame" />
    </node>
</launch>